packages/x-charts/src/internals/domUtils.ts

```diff
@@ -148,13 +148,18 @@ export const getStringSize = (text: string | number, style: React.CSSProperties
       stringCache.cacheCount += 1;
     }
 
-    if (domCleanTimeout) {
-      clearTimeout(domCleanTimeout);
-    }
-    domCleanTimeout = setTimeout(() => {
-      // Limit node cleaning to once per render cycle
+    if (process.env.NODE_ENV === 'test') {
+      // In test environment, we clean the measurement span immediately
       measurementSpan.textContent = '';
-    }, 0);
+    } else {
+      if (domCleanTimeout) {
+        clearTimeout(domCleanTimeout);
+      }
+      domCleanTimeout = setTimeout(() => {
+        // Limit node cleaning to once per render cycle
+        measurementSpan.textContent = '';
```

Comment:
Do we need to clean it? What's the problem of not cleaning it?

Reply:
Else the value is there in the tests when we check for a label value for example. Ok tests we want to immediately remove it as to not pollute test render

Reply:
This line is in the `else` clause, shouldn't that run when we aren't in a test environment?

Reply:
Ah nvm. I got this wrong. This behaviour didn't change. Not sure what is the reasoning for cleaning it. @alexfauquette

Reply:
The comment from this nice guy might help you ;)

https://github.com/mui/mui-x/pull/14042#issuecomment-2285779789

From what I remembered, the DOM was not cleared from one test to the other, so you had an inivisible span in your test because if you search for the last tick label of the axis, it's probably both in the ticklabel, and the measurement span.

Reply:
My question is: why do we need to clean it outside a test environment? 

Changing `textContent` causes the browser to layout, so we might be able to squeeze some more performance out of this method. No need to change it in this PR, though. Just wanted to know if there are any reasons, so I can assess whether it's useful to investigate.

Reply:
outside of test environement, I don't see reasons